archunit/src/test/java/com/tngtech/archunit/testutil/assertion/DependenciesAssertion.java

```diff
@@ -74,6 +75,19 @@ public DependenciesAssertion containOnly(Class<?> expectedOrigin, Class<?> expec
         return this;
     }
 
+    public DependenciesAssertion containOnlyTargetClasses(Class<?>... targetClasses) {
+        assertThat(actual)
+                .extracting(new Extractor<Dependency, Class<?>>() {
+                    @Override
+                    public Class<?> extract(Dependency dependency) {
+                        return dependency.getTargetClass().reflect();
```

Comment:
Maybe it would be better to use `assertThat(javaClass).matches(reflectionClass)`? Because `reflect()` only goes for the name of the `JavaClass` anyway to create the `Class<?>`, no matter if the other attributes of the `JavaClass` matches. If you only want to go for the name (e.g. to not distinguish stubs and imported classes) I think it would be clearer to really only compare the names here then (because that makes it clear that only the names are relevant).

Reply:
When collecting all dependencies, `containOnlyTargetClasses` doesn't make sense anymore, because the result might depend on JRE implementation details.
I've therefore removed this function again.